static/app/views/insights/common/components/chart.tsx

```diff
@@ -143,6 +145,11 @@ function Chart({
   const pageFilters = usePageFilters();
   const {start, end, period, utc} = pageFilters.selection.datetime;
 
+  const chartHeightInContext = useContext(ChartHeightContext);
+  // override the height parameter if the chart height is re-specified in the context. for example, this may happen
+  // when the chart is expanded in full-screen.
+  const height = chartHeightInContext !== undefined ? chartHeightInContext : chartHeight;
```

Comment:
You can probably combine all of this too
```suggestion
  const height = useContext(ChartHeightContext) ?? chartHeight;
```